src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -764,15 +763,168 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
+            break;
+        }
+
+        case NI_SSE2_CompareLessThanOrderedScalar:
+        case NI_SSE2_CompareLessThanUnorderedScalar:
+        {
+            assert(baseType == TYP_DOUBLE);
+            op2Reg          = op2->gtRegNum;
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+
+            emit->emitIns_R_R(ins, emitTypeSize(TYP_SIMD16), op2Reg, op1Reg);
+            emit->emitIns_R(INS_seta, EA_1BYTE, targetReg);
+            emit->emitIns_R_R(INS_movzx, EA_1BYTE, targetReg, targetReg);
+            break;
+        }
+
+        case NI_SSE2_CompareLessThanOrEqualOrderedScalar:
+        case NI_SSE2_CompareLessThanOrEqualUnorderedScalar:
+        {
+            assert(baseType == TYP_DOUBLE);
+            op2Reg          = op2->gtRegNum;
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+
+            emit->emitIns_R_R(ins, emitTypeSize(TYP_SIMD16), op2Reg, op1Reg);
+            emit->emitIns_R(INS_setae, EA_1BYTE, targetReg);
+            emit->emitIns_R_R(INS_movzx, EA_1BYTE, targetReg, targetReg);
+            break;
+        }
+
+        case NI_SSE2_CompareNotEqualOrderedScalar:
+        case NI_SSE2_CompareNotEqualUnorderedScalar:
+        {
+            assert(baseType == TYP_DOUBLE);
+            op2Reg             = op2->gtRegNum;
+            instruction ins    = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+            regNumber   tmpReg = node->GetSingleTempReg();
+
+            emit->emitIns_R_R(ins, emitTypeSize(TYP_SIMD16), op1Reg, op2Reg);
+            emit->emitIns_R(INS_setpe, EA_1BYTE, targetReg);
+            emit->emitIns_R(INS_setne, EA_1BYTE, tmpReg);
+            emit->emitIns_R_R(INS_or, EA_1BYTE, tmpReg, targetReg);
+            emit->emitIns_R(INS_setne, EA_1BYTE, targetReg);
+            emit->emitIns_R_R(INS_movzx, EA_1BYTE, targetReg, targetReg);
+            break;
+        }
+
+        case NI_SSE2_ConvertScalarToVector128Double:
+        {
+            assert(baseType == TYP_INT || baseType == TYP_LONG || baseType == TYP_FLOAT);
+            assert(op1 != nullptr);
+            assert(op2 != nullptr);
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+            genHWIntrinsic_R_R_RM(node, ins);
+            break;
+        }
+
+        case NI_SSE2_ConvertScalarToVector128Int64:
+        case NI_SSE2_ConvertScalarToVector128UInt64:
+        {
+            assert(baseType == TYP_LONG || baseType == TYP_ULONG);
+            assert(op1 != nullptr);
+            assert(op2 == nullptr);
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+            // TODO-XArch-CQ -> use of type size of TYP_SIMD16 leads to
+            // instruction register encoding errors for SSE legacy encoding
+            emit->emitIns_R_R(ins, emitTypeSize(baseType), targetReg, op1Reg);
+            break;
+        }
+
+        case NI_SSE2_ConvertScalarToVector128Single:
```

Comment:
nit: This path could be merged with `ConvertScalarToVector128Double` above. The only difference is the `baseType` assertion.

Reply:
Will do